ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -198,8 +198,6 @@ private void validateCloudMetadata(Map<String, CloudBlobMetadata> cloudBlobMetad
   private long getOperationTime(CloudBlobMetadata metadata, long currentTimeStamp) {
     if (isBlobDeleted(metadata)) {
       return metadata.getDeletionTime();
-    } else if (isBlobExpired(metadata, currentTimeStamp)) {
-      return metadata.getExpirationTime();
```

Comment:
What problem were these lines causing?

Reply:
These lines were not causing any problems. But based on how the operation time is calculated during replication between Ambry data nodes, this logic is inconsistent with that.

In PersistentIndex::findKey() which is called by findEntriesSince of BlobStore, the ttl entry doesn't change the operation time of the entry being replicated. In fact, in order to replicate a TTL_Entry record, the previous PUT record is found, and its ttl flag is set. Hence the operation time is not TTL update's time, rather its the previous PUT time. (Source: PersistentIndex::line609).